File: fund-tracker/frontend/src/types/scenario.ts
export interface ScenarioHorizon {
  horizon_years: number;
  base_cagr: number;
  bull_cagr: number;
  bear_cagr: number;
  sigma_annual: number | null;
  base_value_per_1000: number;
  bull_value_per_1000: number;
  bear_value_per_1000: number;
  notes: string | null;
  generated_at: string;
}

export interface ScenariosData {
  fund_id: number;
  disclaimer: string;
  projections: ScenarioHorizon[];
}

File: fund-tracker/backend/app/schemas/scenario.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional


class ScenarioHorizonSchema(BaseModel):
    horizon_years: int
    base_cagr: float
    bull_cagr: float
    bear_cagr: float
    sigma_annual: Optional[float] = None
    base_value_per_1000: float
    bull_value_per_1000: float
    bear_value_per_1000: float
    notes: Optional[str] = None
    generated_at: datetime


class ScenariosResponse(BaseModel):
    fund_id: int
    disclaimer: str
    projections: list[ScenarioHorizonSchema]
